x-pack/platform/plugins/shared/fleet/server/services/epm/elasticsearch/transform/install.ts

```diff
@@ -882,12 +896,28 @@ const getLegacyTransformNameForInstallation = (
   return `${installablePackage.name}.${folderName}-${filename}${suffix ? '-' + suffix : ''}`;
 };
 
+// If destinationIndex is defined, use it to derive the data type depending on if it matches one of the known
+// patterns with preset component templates. Defaults to `logs` type if no destinationIndex is provided.
+const getTransformDestinationDataType = (
```

Comment:
Suggestions to add some tests for this function.

Reply:
added some unit tests for this helper. note that the actual usage of all the services in this file is covered in `transforms.test.ts`

Reply:
Thanks!